Dockerfile

```diff
@@ -20,108 +26,88 @@ ENV AIIDALAB_DEFAULT_GIT_BRANCH master
-    python3-pip           \
-  && rm -rf /var/lib/apt/lists/*
-
-# Dependencies needed for Jupyter Lab.
-RUN apt-get update && apt-get install -y \
-     nodejs                \
-     npm                   \
-  && rm -rf /var/lib/apt/lists/*
-
-# Install ngrok to be able to proxy AiiDA RESTful API server.
-# Currently not used by the home app, but used in tutorials
-RUN wget --quiet -P /tmp/ \
-  https://bin.equinox.io/a/dnxFaDKQgP4/ngrok-2.3.35-linux-amd64.zip \
-  && unzip /tmp/ngrok-2.3.35-linux-amd64.zip \
-  && mv ./ngrok /usr/local/bin/ \
-  && rm -f /tmp/ngrok-2.3.35-linux-amd64.zip
-
-# Get recent version of pip (needed for `pip cache` command).
-# New pip executable is installed into /usr/local/bin
-RUN /usr/bin/pip3 install --upgrade pip
-
-# Jupyter dependencies installed into system python environment
-# which runs the jupyter notebook server.
-COPY requirements-server.txt .
-RUN /usr/local/bin/pip install -r /opt/requirements-server.txt \
-    && /usr/local/bin/pip cache purge
+     build-essential        \
+&& rm -rf /var/lib/apt/lists/*
 
-# Install and enable appmode.
-RUN git clone https://github.com/oschuett/appmode.git && cd appmode && git reset --hard v0.8.0
-COPY gears.svg ./appmode/appmode/static/gears.svg
-RUN /usr/local/bin/pip install ./appmode
-RUN /usr/local/bin/jupyter nbextension     enable --py --sys-prefix appmode
-RUN /usr/local/bin/jupyter serverextension enable --py --sys-prefix appmode
-
-# Install jupyterlab theme (takes about 4 minutes and 10 seconds).
-#WORKDIR /opt/jupyterlab-theme
-#RUN git clone https://github.com/aiidalab/jupyterlab-theme && \
-#    cd jupyterlab-theme && \
-#     npm install && \
-#     npm run build && \
-#     npm run build:webpack && \
-#     npm pack ./ && \
-#     /usr/local/bin/jupyter labextension install *.tgz && \
-#    cd ..
-
-## Configure user environment
-
-# Install some useful packages that are not available on PyPi.
-RUN conda install --yes -c conda-forge \
-  openbabel==3.1.1 \
-  rdkit==2021.09.2 \
-  && conda clean --all
-
-# Install AiiDAlab Python packages into user conda environment and populate reentry cache.
+# Install the shared requirements.
 COPY requirements.txt .
-ARG extra_requirements
-RUN pip install --upgrade pip
-RUN pip install -r requirements.txt $extra_requirements
-RUN reentry scan
+RUN mamba install --yes \
+     --file requirements.txt && \
+     mamba clean -all -f -y && \
+     fix-permissions "${CONDA_DIR}" && \
+     fix-permissions "/home/${NB_USER}"
+
+# Pin shared requirements in the base environemnt.
+RUN cat requirements.txt | xargs -I{} conda config --system --add pinned_packages {}
 
 # Configure pip to use requirements file as constraints file.
-RUN conda env config vars set PIP_CONSTRAINT=/opt/requirements.txt
+ENV PIP_CONSTRAINT=/opt/requirements.txt
+
+# TODO: should be converted to a conda package
+# ARG aiidalab_version=aiida-2.0
+RUN pip install --quiet --no-cache-dir \
+     # TODO: switch to release version
```

Comment:
- [ ] Switch to release version after release of https://github.com/aiidalab/aiidalab/pull/301